gbpservice/nfp/configurator/agents/loadbalancer_v1.py

```diff
@@ -0,0 +1,655 @@
+
+
+class LBaasRpcSender(Filter):
+    """Agent side of the Agent to Plugin RPC API."""
+
+    API_VERSION = '2.0'
+    # history
+    #   1.0 Initial version
+    #   2.0 Generic API for agent based drivers
+    #       - get_logical_device() handling changed on plugin side;
+    #       - pool_deployed() and update_status() methods added;
+
+    def __init__(self, sc=None):
+        self.sc = sc
+        pass
+
+    def _notification(self, data):
+        LOG.info("Sending notification: %s" % data)
+        event = self.sc.new_event(
+            id='NOTIFICATION_EVENT', key='NOTIFICATION_EVENT', data=data)
+        self.sc.poll_event(event)
+
+    def get_logical_device(self, pool_id, context=None):
+        # Call goes to filter library
+        return self.call(
+            context,
+            self.make_msg(
+                'get_logical_device',
+                pool_id=pool_id
+            )
+        )
+
+    def update_status(self, obj_type, obj_id, status):
+        msg = {'receiver': lb_constants.NEUTRON,
+               'resource': lb_constants.SERVICE_TYPE,
+               'method': 'update_status',
+               'kwargs': [{'obj_type': obj_type,
+                           'obj_id': obj_id,
+                           'status': status}]
+               }
+        self._notification(msg)
+
+    def update_pool_stats(self, pool_id, stats):
+        LOG.info("[LbaaSRpcSender]: Update pool stats called")
+        msg = {'receiver': lb_constants.NEUTRON,
+               'resource': lb_constants.SERVICE_TYPE,
+               'method': 'update_pool_stats',
+               'kwargs': [{'pool_id': pool_id,
+                           'stats': stats}]
+               }
+        self._notification(msg)
+
+
+"""Implements APIs invoked by configurator for processing RPC messages.
+
+RPC client of configurator module receives RPC messages from REST server
+and invokes the API of this class. The instance of this class is registered
+with configurator module using register_service_agent API. Configurator module
+identifies the service agent object based on service type and invokes ones of
+the methods of this class to configure the device.
+
+"""
+
+
+class LBaaSRpcManager(agent_base.AgentBaseRPCManager):
+    def __init__(self, sc, conf):
+        """Instantiates child and parent class objects.
+
+        Passes the instances of core service controller and oslo configuration
+        to parent instance inorder to provide event enqueue facility for batch
+        processing event.
+
+        :param sc: Service Controller object that is used for interfacing
+        with core service controller.
+        :param conf: Configuration object that is used for configuration
+        parameter access.
+
+        """
+
+        super(LBaaSRpcManager, self).__init__(sc, conf)
+
+    def create_vip(self, context, vip):
+
+        arg_dict = {'context': context,
+                    'vip': vip,
+                    'serialize': True,
```

Comment:
This will not work. Why are the keys "serialize" and "binding_key" passed inside the data dict? They should be arguments to new_event() method right?